Problem with Windows preprocessor definition on Linux

Hi,

**OS**: Linux
**Plus version**: latest

When compiling some code using the Plus library headers, we got a strange compilation issue related to boost inclusion (boost asio is used in our framework as well):

```
/home/user/.conan/data/boost/1.72.0/_/_/package/15e881cf4fa352877c6f626d88a3b9471860ac90/include/boost/asio/basic_socket.hpp:555:46: error: invalid token at start of a preprocessor expression [clang-diagnostic-error]
  && (!defined(_WIN32_WINNT) || _WIN32_WINNT < 0x0603) 
```

The issue is due to the fact that somewhere `_WIN32_WINNT` is defined but empty. It happens it is caused by `PlusConfigure.h` included in some of the Plus library headers that defines the variable unconditionnally event if empty on Linux.

--- a/src/PlusConfigure.h.in
+++ b/src/PlusConfigure.h.in
@@ -11,10 +11,12 @@
 #ifndef __PlusConfigure_h
 #define __PlusConfigure_h
 
+#if defined(_WIN32)
 #ifdef _WIN32_WINNT
 #undef _WIN32_WINNT
 #endif
 #define _WIN32_WINNT @RUNTIME_MINIMUM_WINDOWS_VERSION@
+#endif
 
 // disable warnings for sprintf
 #define _CRT_SECURE_NO_WARNINGS

